### PR TITLE
RavenDB-10237

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs [hunk 1]
```diff
@@ -120,7 +120,7 @@ public Task GetObserverDecisions()
                     var res = ServerStore.Observer.ReadDecisionsForDatabase();
                     var json = new DynamicJsonValue{
                         [nameof(ClusterObserverDecisions.LeaderNode)] = Server.ServerStore.NodeTag,
-                        [nameof(ClusterObserverDecisions.Term)] = Server.ServerStore.Engine.CurrentTerm,
+                        [nameof(ClusterObserverDecisions.Term)] = Server.ServerStore.Engine.CurrentLeader.Term,
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs:123]: Possible NRE?

File: src/Raven.Server/Rachis/Elector.cs [hunk 2]
```diff
@@ -41,7 +40,7 @@ public void HandleVoteRequest()
 
                         if (_engine.Log.IsInfoEnabled)
                         {
-                            var election = rv.IsTrialElection ? "Trail" : "Real";
+                            var election = rv.IsTrialElection ? "Trial" : "Real";
```
Comment (ayende) [inline src/Raven.Server/Rachis/Elector.cs:43]: +1

File: src/Raven.Server/Rachis/RachisConsensus.cs [hunk 3]
```diff
@@ -1009,10 +1010,14 @@ private void ValidateElectionTimeout(RachisHello initialMessage)
...
         }
 
-        public unsafe long InsertToLeaderLog(TransactionOperationContext context, BlittableJsonReaderObject cmd,
+        public unsafe long InsertToLeaderLog(TransactionOperationContext context, long term, BlittableJsonReaderObject cmd,
             RachisEntryFlags flags)
         {
             Debug.Assert(context.Transaction != null);
+            
+            if(term != CurrentTerm)
+                throw new ConcurrencyException();
```
Comment (ayende) [inline src/Raven.Server/Rachis/RachisConsensus.cs:1019]: Proper error

File: src/Raven.Server/Rachis/Follower.cs [hunk 4]
```diff
@@ -334,25 +338,21 @@ private void NegotiateWithLeader(TransactionOperationContext context, LogLengthN
...
             {
-                _engine.Log.Info($"{ToString()}: Got a negotiation request for term {negotiation.Term} where our term is {_engine.CurrentTerm}");
+                _engine.Log.Info($"{ToString()}: Got a negotiation request for term {negotiation.Term} where our term is {_term}");
             }
-            if (negotiation.Term < _engine.CurrentTerm)
+            if (negotiation.Term != _term)
             {
                 //  Our leader is no longer a valid one
-                var msg = $"My term is higher then yours {_engine.CurrentTerm} > {negotiation.Term}, so you are no longer a valid leader";
+                var msg = $"My term is different then yours {_term} != {negotiation.Term}, so you are no longer a valid leader";
```
Comment (ayende) [inline src/Raven.Server/Rachis/Follower.cs:346]: fix msg

File: src/Raven.Server/Rachis/CandidateAmbassador.cs [hunk 5]
```diff
@@ -187,7 +187,7 @@ private void Run()
                                     {
                                         var message = $"Candidate ambassador for {_tag}: found election term {rvr.Term:#,#;;0} that is higher than ours {currentElectionTerm:#,#;;0}";
                                         // we need to abort the current elections
-                                        _engine.SetNewState(RachisState.Follower, null, _engine.CurrentTerm, message);
+                                        //_engine.SetNewState(RachisState.Follower, null, _engine.CurrentTerm2, message);
```
Comment (talweiss1982) [inline src/Raven.Server/Rachis/CandidateAmbassador.cs]: Can we find a more meaningful name than CurrentTerm2?

**Pull request merged**